src/kernel/src/fs/mod.rs

```diff
@@ -284,18 +326,29 @@ impl Fs {
         }
 
         // Get data.
-        let data = if size == 0 {
-            if com & IOC_IN != 0 {
-                todo!("ioctl with IOC_IN");
-            } else if com & IOC_OUT != 0 {
-                todo!("ioctl with IOC_OUT");
-            }
-
-            &[]
+        let mut data = if size == 0 {
+            &mut []
         } else {
-            todo!("ioctl with size != 0");
+            if com & IOC_VOID != 0 {
+                todo!("ioctl with com & IOC_VOID != 0");
+            } else {
+                if size > SYS_IOCTL_SMALL_SIZE {
+                    //malloc here
+                    todo!("ioctl with size != 0 and com & IOC_VOID == 0 and size > SYS_IOCTL_SMALL_SIZE")
+                } else {
+                    bytemuck::cast_slice_mut(&mut smalldata)
+                }
```

Comment:
Where did you get this from? On the 9.0 kernel it will always `malloc` here.

Reply:
I don't remember, but you are correct. Sorry